Aptoide/appcoins-wallet-android
# [WAL-53] Gamification highlight remaining integration

**What does this PR do?**

Fixed some bugs with previous gamification highlight implementation
Added Gamification Highlight to topup
Changed rule to always show level up expcept if max level (max level layout will be implemented later)
Fixed bug with back button override on AdyenPaymentFragment when user has a stored card

**Database changed?**

    No

**Where should the reviewer start?**

PaymentMethodsPresenter.kt
TopUpFragmentPresenter.kt

**How should this be manually tested?**

Check different use cases in both top up and iab
Check rotation
Check switching wallets while on iab and check if the info updates 

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/WAL-53

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)




**Code Review Checklist**
